### PR TITLE
Add get_full_qualified_names() to std, js and cpp domains.

### PR DESCRIPTION
**Pull request by tk0miya**
refs: #3630

### CODE REVIEW

File: tests/test_domain_cpp.py [hunk 1]
```diff
@@ -13,6 +13,8 @@
 
 from six import text_type
 import pytest
+from mock import Mock
+from docutils import nodes
```
Comment (shimizukawa) [inline tests/test_domain_cpp.py]: What's this?

**Review by shimizukawa — APPROVED** [review]
LGTM with nits.
(Actually, I just zapped whole diff)

File: tests/test_domain_cpp.py (hunk 1 above)
Comment (tk0miya) [reply tests/test_domain_cpp.py]: Oh, it's my mistake. Thanks, removed.

**Review by jakobandersen — APPROVED** [review]
Looks good, it should keep the current behaviour. When I find time I'll look into what is needed to support more complex declarations.

**Comment by tk0miya** [comment]
Merged, Thank you for quick reviewing!

**Pull request merged**